src/views/pages/ManagePeople/index.vue

```diff
@@ -52,81 +54,68 @@ export default Vue.extend({
       this.searched = searchByName(this.people, this.search);
     },
   },
-  created: function created() {
-    this.searched = this.people;
+  created: async function created() {
+    const self = this;
+
+    // get people types
+    const personCategories: any[] = [];
+    await axios.get('http://boresha.tech:9090/people/categories')
+      .then((response: any) => {
+        response.data.map((category: any) => {
+          personCategories.push(category.name);
+        });
+      })
+      .catch((error: any) => {
+        console.log(error.message);
+        self.error = error.message;
+      });
+
+    // get all people
+    personCategories.map((category: string) => {
+      // get all people of particular category
+      axios.get('http://boresha.tech:9090/people/' + category)
+        .then((response: any) => {
+          // construct each person
+          response.data.map((person: any) => {
+            // construct full name
+            const fullName: string =
+              `${person.firstName || ''} ${person.middleName || ''} ${person.lastName || ''}`;
+
+            // construct phone number
+            let fullPhone: string = '';
+            if (person.phoneCountry) {
+              fullPhone += `+${person.phoneCountry}`;
+            }
+            if (person.phoneArea) {
+              fullPhone += ` (${person.phoneArea})`;
+            }
+            if (person.phoneNumber) {
+              fullPhone += ` ${person.phoneNumber.slice(0, 3)}-${person.phoneNumber.slice(3)}`;
```

Comment:
I'm not totally sure what you mean about the phone library not fitting our use case, you can specify which country it should be yourself and validate that they have provided an area code in some way. It would offer a lot more flexibility in how people put a number down, and give us a reliable way to parse it. There are so many edge cases in a problem as complex as phone numbers that would lead to bugs down the road if we try to do it ourselves in 8 lines of code

Reply:
Well the one that one wasn't fitting is because all the fields for this phone number are optional, I didn't see an easy way to pass a bunch of fields that could be null.